packages/loopring_v3/contracts/aux/bridge/Bridge.sol

```diff
@@ -0,0 +1,605 @@
+                    tokens[tokenIdx].token = token;
+                    tokens[tokenIdx].tokenID = _getTokenID(token);
+                    numDistinctTokens++;
+                }
+            }
+            tokens[tokenIdx].amount = tokens[tokenIdx].amount.add(deposits[i].amount);
+            deposits[i].token = address(tokens[tokenIdx].tokenID);
+
+            transfers[i].owner = deposits[i].owner;
+            transfers[i].tokenID = tokens[tokenIdx].tokenID;
+            transfers[i].amount = deposits[i].amount;
+        }
+
+        // Do a normal deposit per token
+        for(uint i = 0; i < numDistinctTokens; i++) {
+            if (tokens[i].token == address(0)) {
+                require(tokens[i].amount == msg.value, "INVALID_ETH_DEPOSIT");
+            }
+            _deposit(from, tokens[i].token, uint96(tokens[i].amount));
+        }
+
+        // Store the transfers so they can be processed later
+        _storeTransfers(transfers);
+    }
+
+    function beforeBlockSubmission(
+        bytes memory   txsData,
+        bytes calldata callbackData
+        )
+        external
+        onlyFromExchangeOwner
+    {
+        BridgeOperations memory operations = abi.decode(callbackData, (BridgeOperations));
+
+        Context memory ctx = Context({
+            domainSeparator: DOMAIN_SEPARATOR,
+            accountID: accountID,
+            txIdx: 0
+        });
+
+        _processTransfers(ctx, operations.transferOperations, txsData);
+        _processCalls(ctx, operations, txsData);
+
+        // Make sure we have consumed exactly the expected number of transactions
+        require(txsData.length == ctx.txIdx * ExchangeData.TX_DATA_AVAILABILITY_SIZE, "INVALID_NUM_TXS");
+    }
+
+    // Allows withdrawing from pending transfers that are at least MAX_AGE_PENDING_TRANSFERS old.
+    function withdrawFromPendingTransfers(
+        uint                              batchID,
+        InternalBridgeTransfer[] calldata transfers,
+        uint[]                   calldata indices
+        )
+        external
+    {
+        // Check if withdrawing from these transfers is possible
+        bytes32 hash = _hashTransfers(batchID, transfers);
+        require(_arePendingTransfersTooOld(hash), "TRANSFERS_NOT_TOO_OLD");
+
+        for (uint i = 0; i < indices.length; i++) {
+            uint idx = indices[i];
+
+            require(!withdrawn[hash][idx], "ALREADY_WITHDRAWN");
+            withdrawn[hash][idx] = true;
+
+            address tokenAddress = exchange.getTokenAddress(transfers[idx].tokenID);
+
+            _transferOut(
+                tokenAddress,
+                transfers[idx].amount,
+                transfers[idx].owner
+            );
+        }
+    }
+
+    // Can be used to withdraw funds that were already deposited to the bridge,
+    // but need to be returned to be able to withdraw from old pending transfers.
+    function forceWithdraw(address[] calldata tokens)
+        external
```

Comment:
It's to allow withdrawing directly from the Bridge for old batched deposit transfers. Because the deposit already happened users can just call `forceWithdraw` to withdraw funds back to the bridge so the withdrawal can be done on L1.

The operator can still schedule the actual forced withdrawal when it wants, so can be made public without any problem.

Reply:
Yep, first bring back funds to the Bridge contract (if nobody else has done so already, any funds already in the Bridge contract can be used) and then withdraw from there.